agent/grpc-external/server.go

```diff
@@ -4,27 +4,56 @@
 package external
 
 import (
+	"context"
+	"fmt"
+	"strings"
 	"time"
 
 	"github.com/armon/go-metrics"
 	middleware "github.com/grpc-ecosystem/go-grpc-middleware"
 	recovery "github.com/grpc-ecosystem/go-grpc-middleware/recovery"
+	"github.com/hashi-derek/grpc-proxy/proxy"
 	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
 	"google.golang.org/grpc/credentials"
 	"google.golang.org/grpc/keepalive"
+	"google.golang.org/grpc/metadata"
+	"google.golang.org/grpc/status"
 
 	"github.com/hashicorp/consul/agent/consul/rate"
 	agentmiddleware "github.com/hashicorp/consul/agent/grpc-middleware"
 	"github.com/hashicorp/consul/tlsutil"
 )
 
+const FORWARDED_METHOD = "hashicorp.consul.resource.ResourceService"
```

Comment:
Perhaps we should do some more careful matching on this, rather than doing a `Contains()` call. Perhaps something like `HasPrefix()` would work well if you ensure a terminator is added at the end of the RPC name (maybe something like a trailing slash? `/`)

For example, if we keep this logic and later-on add in a `hashicorp.consul.resource.ResourceServiceManager`, then it would still be a match and might get forwarded on accident. Setting it to look for a prefix of `hashicorp.consul.resource.ResourceService/` may prevent that overmatch from happening in the future.

Reply:
Very great catch!!